util-taglib/src/com/liferay/taglib/ui/MessageTag.java

```diff
@@ -58,11 +58,17 @@ public int doEndTag() throws JspException {
 				if (!_localizeKey) {
 					value = _key;
 				}
-				else if (_escape) {
+				else if (_escape &&
+						 !LanguageUtil.isValidLanguageKey(
+							 resourceBundle.getLocale(), _key)) {
+
 					value = HtmlUtil.escape(
 						LanguageUtil.get(resourceBundle, _key));
```

Comment:
This doesn't make sense. Why lookup `_key` in the `resourceBundle` if the key is not a valid language key.